src/internal/v1api/generator.go

```diff
@@ -0,0 +1,26 @@
+package v1api
+
+import (
+	"io/fs"
+
+	"registry-stable/internal/files"
+)
+
+// Generator is responsible for generating the responses used by v1 registry APIs.
+// This should take information from the current state of the registry and generate the responses
+// to be served by the API.
+// For more information on the API see the following documentation:
+// - https://opentofu.org/docs/internals/module-registry-protocol/
+// - https://opentofu.org/docs/internals/provider-registry-protocol
+type Generator struct {
+	// DestinationDir is the directory to write the generated responses to.
+	DestinationDir string
+
+	// TODO: Maybe combine all the fs.FS stuff and FileWriter stuff into a single interface to be consumed easier?
```

Comment:
Honestly, to me this looks like a lot of needless abstraction.

We're operating on the file system, I don't see a reason to mock the file system. Just make all functions accept the directory they should operate on and have the Generator contain the relevant file paths, imo.

Reply:
Agreed

Reply:
I mainly wanted to mock it so i can test with an in-mem filesystem, and I thought it would be easier to do it from the get-go rather than shoehorning it in later. But I agree, its maybe a bit too much.